packages/graphql/tests/schema/fulltext.test.ts

```diff
@@ -168,6 +174,8 @@ describe("@fulltext schema", () => {
               movies(fulltext: MovieFulltext, options: MovieOptions, where: MovieWhere): [Movie!]!
```

Comment:
Now, if I'm being honest I'm not 100% what the language allows, but if we could do the following it would be amazing:

```suggestion
              movies(fulltext: MovieFulltext @deprecated(reason: "Use root Query level full-text fields instead"), options: MovieOptions, where: MovieWhere): [Movie!]!
```

And then also mark the `MovieFulltext` type as deprecated as well:

```gql
input MovieFulltext @deprecated(reason: "Use root Query level full-text fields instead") {
    MovieDescription: MovieMovieDescriptionFulltext
    MovieTitle: MovieMovieTitleFulltext
}
```

If you want to get _super_ creative:

```gql
input MovieFulltext @deprecated(reason: "Use root Query level full-text fields instead") {
    MovieDescription: MovieMovieDescriptionFulltext @deprecated(reason: "Use Query.moviesFulltextMovieDescription instead")
    MovieTitle: MovieMovieTitleFulltext @deprecated(reason: "Use Query.moviesFulltextMovieTitle instead")
}
```

That last one is probably more trouble than it's worth!

Perhaps also worth deprecating the other input types:

```gql
input MovieMovieDescriptionFulltext @deprecated(reason: "Use root Query level full-text fields instead") {
    phrase: String!
}

input MovieMovieTitleFulltext @deprecated(reason: "Use root Query level full-text fields instead") {
    phrase: String!
}
```

Choose a reason you feel is appropriate! 🙂 Also, unsure how much of the above is even possible, GraphQL can be a bit picky.

Reply:
Unfortunately, I believe I tried most of these when I was working on this PR. The `@deprecated` directive can only be used on fields (but not input fields) so doesn't work anywhere we can put it.

The old implementation for full-text just used the normal return object so it's not even possible to deprecate the fields on there as it would deprecate them for all queries (not just full-text)

Reply:
Damn, I see this now, that's disappointing! Seems like a shortcoming of the directive to me...